sys/random/xorshift.c

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * Code taken from Wikipedia (https://en.wikipedia.org/wiki/Xorshift)
+ */
+
+ /**
+ * @ingroup sys_random
+ * @{
+ * @file
+ *
+ * @brief   Xorshift random number generator implementation
+ *
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+
+#include "random.h"
+
+/* The state word must be initialized to non-zero */
+uint32_t xorshift32(uint32_t *state)
```

Comment:
minor: why did you change the signature from the wikipedia example?
In the example it is:
```c
uint32_t state[static 1]
```

Reply:
I'd never seen that notation before, and as far as I can tell, using a pointer is equivalent, right?

Reply:
but you kept the same notation at
https://github.com/RIOT-OS/RIOT/pull/7545/files#diff-82a7ba313c18943a85a93c8005a1795eR41

my google-fu tells me that compilers might do some optimizations when using this notation (I also did not see that notation before). However, I would prefer the pointer syntax.

Reply:
I'll change the other place.

Reply:
done